### ISSUE TITLE
Manually set active station is ignored with "backward legs, splays" stations policy

### ISSUE THREAD

**Issue by rsevero** [marcocorvi/topodroid#79]
**Describe the bug**
When using "backward legs, splays" stations policy manually set active station (making a station green) is ignored

**To Reproduce**
1. set stations policy to "backward legs, splays" 
2. Create some shots
3. Manually set the active station as some station different from the last
4. Make a measurement for a splay

**Expected behavior**
The new splays origin should be the manually choosen station but it is the last station created

**Device (please complete the following information):**
 - Samsung Galaxy S21 FE and Motorola A32
 - Android version 13
 - TopoDroid version 6.2.34-33

**Comment by marcocorvi** [comment]
this is the expected behaviour.
active station is applies only to legs, 
splays assignment is by the  policy

**Comment by rsevero** [comment]
For policy "Splays, forward legs" splay assignement is by "active station"

For policy "backward legs, splays" splay assignement is by "policy"

Besides this difference in behaviour that I believe shouldn't have because of different policies, I really don't understand why splay assignment wouldn't respect "active station". I have always used it. It is really usefull. Yesterday I found out that "backward legs, splays" behaves, in this aspect, differently to "Splays, forward legs".

**Comment by marcocorvi** [comment]
then it must be an error in the policy code.
i wiil check it.

**Comment by marcocorvi** [comment]
the same behaviour was in the policy "first forward-leg then splays"

somehow there was a comment in the code that said it was to be like that,
and bypassed the check of the active station.

i do not remember why i put it,
and i am not sure that adding the check there might be some drawback.

**Comment by rsevero** [comment]
With this file with TopodroidX 6.2.35 and a BRIC4 I have 2 issues:

1. setting active station still ignored, i.e., new splays created from last created station, ignmoring "active station" setting
2. turning off "legs-only refresh" inefective: to see new splays I have to go to shots list and them back to drawing window.

[teste2.zip](https://github.com/marcocorvi/topodroid/files/13840388/teste2.zip)

**Comment by marcocorvi** [comment]
Only bric4 or also distox2 ?

**Comment by rsevero** [comment]
> Only bric4 or also distox2 ?

Again the tests that failed were with a BRIC4. Forgot to test with disto.

**Comment by marcocorvi** [comment]
the wrong behavior could not been reproduced 
v. 6.2.35

**Comment by rsevero** [comment]
Yes. It seems to be working now. Strange. If I manage to get more solid info I wil get back to you. Sorry for the noise.

**Comment by marcocorvi** [comment]
since it "seems" to be working i close the issue.
